src/c-writer.cc

```diff
@@ -3317,9 +3388,13 @@ void CWriter::Write(const LoadExpr& expr) {
   Memory* memory = module_->memories[module_->GetMemoryIndex(expr.memidx)];
 
   Type result_type = expr.opcode.GetResultType();
-  Write(StackVar(0, result_type), " = ", func, "(",
-        ExternalInstancePtr(ModuleFieldType::Memory, memory->name), ", (u64)(",
-        StackVar(0), ")");
+  if (options_.no_sandbox) {
```

Comment:
Given the changes to wasm2c.declarations.nosandbox.c, could this case be eliminated? (I.e. ignore the memory in the actual load function.)